version_service.yaml

```diff
@@ -0,0 +1,244 @@
+        - "application/json"
+      parameters:
+        - $ref: '#/parameters/productParam'
+        - $ref: '#/parameters/applyParam'
+        - $ref: '#/parameters/operatorVersionParam'
+        - $ref: '#/parameters/databaseVersionParam'
+        - $ref: '#/parameters/kubeVersionParam'
+        - $ref: '#/parameters/platformParam'
+        - $ref: '#/parameters/pmmVersionParam'
+        - $ref: '#/parameters/backupVersionParam'
+        - $ref: '#/parameters/namespaceOIDParam'
+        - $ref: '#/parameters/customResourceOIDParam'
+      responses:
+        200:
+          description: "successful operation"
+          schema:
+            $ref: '#/definitions/UpgradeVersionBlock'
+        400:
+          $ref: '#/responses/400ClientError'
+        404:
+          $ref: '#/responses/404NotFound'
+definitions:
+  VersionBlock:
+    type: "object"
+    properties:
+      versions:
+        type: "array"
+        items:
+          $ref: '#/definitions/OperatorBlock'
+  OperatorBlock:
+    type: "object"
+    properties:
+      operator:
+        type: "string"
+        pattern: '^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$'
+      product:
+        type: "string"
+        pattern: '^psmdb-operator|pxc-operator$'
+      matrix:
+        $ref: '#/definitions/VersionMatrix'
+  OperatorVersionBlock:
+    $ref: '#/definitions/VersionBlock'
+  UpgradeVersionBlock:
+    $ref: '#/definitions/VersionBlock'
+  VersionMatrix:
+    type: "object"
+    properties:
+      mongod:
+        $ref: '#/definitions/DatabaseObject'
+      pxc:
+        $ref: '#/definitions/DatabaseObject'
+      pmm:
+        $ref: '#/definitions/PMMObject'
+      proxysql:
+        $ref: '#/definitions/ProxySQLObject'
+      haproxy:
+        $ref: '#/definitions/HAProxyObject'
+      backup:
+        $ref: '#/definitions/BackupObject'
+      operator:
+        $ref: '#/definitions/OperatorObject'
+  DatabaseObject:
+    type: "object"
+    additionalProperties:
+      $ref: '#/definitions/VersionObject'
+  PMMObject:
+    type: "object"
+    additionalProperties:
+      $ref: '#/definitions/VersionObject'
+  ProxySQLObject:
+    type: "object"
+    additionalProperties:
+      $ref: '#/definitions/VersionObject'
+  HAProxyObject:
+    type: "object"
+    additionalProperties:
+      $ref: '#/definitions/VersionObject'
+  BackupObject:
+    type: "object"
+    additionalProperties:
+      $ref: '#/definitions/VersionObject'
+  OperatorObject:
+    type: "object"
+    additionalProperties:
+      $ref: '#/definitions/VersionObject'
+  VersionObject:
+    type: "object"
+    properties:
+      imagepath:
+        type: "string"
+        format: "uri"
+      imagehash:
+        type: "string"
+        pattern: '^sha256:[A-Fa-f0-9]{64}$'
+      status:
+        type: "string"
+        pattern: '^recommended|available|required|disabled$'
```

Comment:
https://github.com/Percona-Lab/percona-version-service/pull/1/commits/0070f420a7bfe0ee9542c7c4e0bf0a845cb032e7
I have used enums, but that leads a few issues with json unmarshal. Had to write statuses in lower case because in json we have it in lowercase (of course we can change it in json to uppercase, but it's kind of spec change and did not look well).
Because of this I had to disable some prototool linters related to enums.